internal/pkg/term/selector/selector_test.go

```diff
@@ -3127,3 +2990,214 @@ func TestTaskSelect_Task(t *testing.T) {
+				m.EXPECT().SelectOne(
+					gomock.Any(), gomock.Any(),
+					gomock.Eq([]string{
+						"./Dockerfile",
+						"backend/my.dockerfile",
+						"frontend/Dockerfile",
+						dockerfilePromptUseCustom,
+						DockerfilePromptUseImage,
+					}),
+					gomock.Any(),
+				).Return("frontend/Dockerfile", nil)
+			},
+			wantedDockerfile: "frontend/Dockerfile",
+		},
+		"prompts user for custom path": {
+			mockFileSystem: func(mockFS afero.Fs) {},
+			mockPrompt: func(m *mocks.Mockprompter) {
+				m.EXPECT().SelectOne(
+					gomock.Any(), gomock.Any(),
+					gomock.Eq([]string{
+						dockerfilePromptUseCustom,
+						DockerfilePromptUseImage,
+					}),
+					gomock.Any(),
+				).Return("Enter custom path for your Dockerfile", nil)
+				m.EXPECT().Get(
+					gomock.Any(),
+					gomock.Any(),
+					gomock.Any(),
+					gomock.Any(),
+				).Return("crazy/path/Dockerfile", nil)
+			},
+			wantedDockerfile: "crazy/path/Dockerfile",
+		},
+		"returns an error if fail to select Dockerfile": {
+			mockFileSystem: func(mockFS afero.Fs) {},
+			mockPrompt: func(m *mocks.Mockprompter) {
+				m.EXPECT().SelectOne(
+					gomock.Any(),
+					gomock.Any(),
+					gomock.Eq([]string{
+						dockerfilePromptUseCustom,
+						DockerfilePromptUseImage,
+					}),
+					gomock.Any(),
+				).Return("", errors.New("some error"))
+			},
+			wantedErr: fmt.Errorf("select Dockerfile: some error"),
+		},
+		"returns an error if fail to get custom Dockerfile path": {
+			mockFileSystem: func(mockFS afero.Fs) {
+				_ = afero.WriteFile(mockFS, "Dockerfile", []byte("FROM nginx"), 0644)
+			},
+			mockPrompt: func(m *mocks.Mockprompter) {
+				m.EXPECT().SelectOne(
+					gomock.Any(), gomock.Any(),
+					gomock.Eq([]string{
+						"./Dockerfile",
+						dockerfilePromptUseCustom,
+						DockerfilePromptUseImage,
+					}),
+					gomock.Any(),
+				).Return("Enter custom path for your Dockerfile", nil)
+				m.EXPECT().Get(
+					gomock.Any(),
+					gomock.Any(),
+					gomock.Any(),
+					gomock.Any(),
+				).Return("", errors.New("some error"))
+			},
+			wantedErr: fmt.Errorf("get custom Dockerfile path: some error"),
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			// GIVEN
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			p := mocks.NewMockprompter(ctrl)
+			fs := &afero.Afero{Fs: afero.NewMemMapFs()}
+			tc.mockFileSystem(fs)
+			tc.mockPrompt(p)
+
+			sel := localFileSelector{
+				prompt: p,
+				fs:     fs,
+			}
+
+			mockPromptText := "prompt"
+			mockHelpText := "help"
+
+			// WHEN
+			dockerfile, err := sel.Dockerfile(
+				mockPromptText,
+				mockPromptText,
+				mockHelpText,
+				mockHelpText,
+				func(v interface{}) error { return nil },
+			)
+
+			// THEN
+			if tc.wantedErr != nil {
+				require.EqualError(t, err, tc.wantedErr.Error())
+			} else {
+				require.Equal(t, tc.wantedDockerfile, dockerfile)
+			}
+		})
+	}
+}
+
+func TestLocalFileSelector_listDockerfiles(t *testing.T) {
```

Comment:
Similarly, these test cases are (mostly) copied from https://github.com/aws/copilot-cli/blob/e7fadd11e2bea021a80c977fa210d3fe032b142a/internal/pkg/workspace/workspace_test.go#L1477.

Except for `"exclude Dockerfiles in parent directories of the working dir"` which is a new test case.